src/private/MDMPerformerGroup.m

```diff
@@ -42,35 +43,41 @@ - (instancetype)initWithTarget:(id)target scheduler:(MDMScheduler *)scheduler {
   if (self) {
     _target = target;
     _scheduler = scheduler;
-
     _performerInfos = [NSMutableArray array];
     _performerClassNameToPerformerInfo = [NSMutableDictionary dictionary];
+    _performerPlanNameToPerformerInfo = [NSMutableDictionary dictionary];
     _activePerformers = [NSMutableSet set];
   }
   return self;
 }
 
-- (void)addPlan:(id<MDMPlan>)plan trace:(MDMTrace *)trace {
-  [trace.committedPlans addObject:plan];
-
+- (void)addPlan:(nonnull id<MDMPlan>)plan to:(nonnull id)target trace:(MDMTrace *)trace {
   BOOL isNew = NO;
-  id<MDMPerforming> performer = [self performerForPlan:plan isNew:&isNew];
-
-  if (isNew) {
-    [trace.createdPerformers addObject:performer];
-
-    for (id<MDMTracing> tracer in self.scheduler.tracers) {
-      if ([tracer respondsToSelector:@selector(didCreatePerformer:for:)]) {
-        [tracer didCreatePerformer:performer for:self.target];
-      }
-    }
-  }
-
+  id<MDMPerforming> performer = [self findOrCreatePerformerForPlan:plan trace:trace isNew:&isNew];
+  [self notifyPlanAdded:plan to:target trace:trace isNew:isNew performer:performer];
   if ([performer respondsToSelector:@selector(addPlan:)]) {
     [(id<MDMPlanPerforming>)performer addPlan:plan];
   }
 }
 
+- (void)addPlan:(nonnull id<MDMNamedPlan>)plan named:(nonnull NSString *)name to:(nonnull id)target trace:(MDMTrace *)trace {
+  // remove first
+  BOOL isNew = NO;
+  MDMPerformerInfo *performerInfo = [self findOrCreatePerformerInfoForNamedPlan:plan named:name trace:trace isNew:&isNew];
+  [self removePlanNamed:name from:target withPerformerInfo:performerInfo];
+  [trace.committedRemovePlans addObject:plan];
+  // then add
+  id<MDMPerforming> performer = performerInfo.performer;
+  [self notifyPlanAdded:plan to:target trace:trace isNew:isNew performer:performer];
+  if ([performer respondsToSelector:@selector(addPlan:named:)]) {
+    [(id<MDMNamedPlanPerforming>)performer addPlan:plan named:name];
+  }
+}
+
+- (void)removePlanNamed:(nonnull NSString *)name from:(nonnull id)target {
+  [self removePlanNamed:name from:target withPerformerInfo:self.performerPlanNameToPerformerInfo[name]];
```

Comment:
Still looking into this, but removing the performer in `removePlanNamed` results in this unit test failing:
```
func testAddAndRemoveTheSameNamedPlan() {
    let scheduler = Scheduler()

    scheduler.addPlan(firstViewTargetAlteringPlan, named: "name_one", to: target)
    scheduler.removePlan(named: "name_one", from: target)

    XCTAssertTrue(target.text! == "removePlanInvokedaddPlanInvokedremovePlanInvoked")
  }
```